homeassistant/components/smarttub/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow to configure the SmartTub integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_SCAN_INTERVAL
+from homeassistant.core import callback
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN  # pylint: disable=unused-import
+from .controller import SmartTubController
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SmartTubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """SmartTub configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            controller = SmartTubController(self.hass)
+            account_id = await controller.get_account_id(
+                user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+            )
+            if account_id is not None:
+                await self.async_set_unique_id(account_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input[CONF_EMAIL], data=user_input
+                )
+
+            errors["base"] = "invalid_auth"
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
```

Comment:
It doesn't look like you're listening on this, so it'll only take after restarting HA/reloading the integration

Reply:
Is this the correct way to do it? https://github.com/home-assistant/core/pull/37775/commits/d813ef110b64807fd3b71972296c2bf57cf3b761

Reply:
Is this the correct way to do it? https://github.com/home-assistant/core/pull/37775/commits/d813ef110b64807fd3b71972296c2bf57cf3b761